### ISSUE TITLE
Inference from python

### ISSUE THREAD

**Issue by faroit** [facebookresearch/demucs#376]
what is the simpliest way with a pip installed demucs to infer estimates without intermediate files to be stored?

```
from demucs import pretrained, apply
model = pretrained.get_model(name="mdx_extra_q")
model.eval()

def infer_drums_demucs(mix_np):
    # mix_np is of shape (channels, time)
    mix = torch.tensor(mix_np).float()
    ref = mix.mean(0)
    mix = (mix - ref.mean()) / ref.std()
    sources = apply.apply_model(model, mix[None], split=False, overlap=0.25, progress=True)[0]
    sources = sources * ref.std() + ref.mean()
    return sources[model.sources.index('vocals')].detach().cpu().numpy()
```

Does this look correct?

**Comment by adefossez** [comment]
That's correct yes, except you don't really need the normalization anymore, as Demucs uses normalization builtin the model now, but it doesn't hurt to have it!

Also your function is called `infer_drums` but actually return vocals ^^

**Comment by faroit** [comment]
@adefossez thanks. Now that this is verified, I wonder if there is something else that might be wrong when comparing against open-unmix drums. What I am aiming to do is to reproduce the metrics of the [MSG paper](https://interactiveaudiolab.github.io/project/msg.html) from @noahschaffer.

What I observed is that open-unmix drums performs better in the F1 onset_strength metric than demucsv3. And also better in MSE on magnitude spectrograms...an just slightly worse in time-domain mse – that can clearly not be correct! Any ideas?

https://colab.research.google.com/drive/1OW-Es5JkvTcfCbtx5VtGl0tXwcwomSwi?usp=sharing

These are the paper results I wanted to reproduce:

<img width="392" alt="image" src="https://user-images.githubusercontent.com/72940/189858940-085c562d-0517-4950-ab21-3d4dbf9e60b5.png">

**Comment by adefossez** [comment]
For long sequences you need to pass `split=True`. I would also try `shift=1` but it shouldn't change much things. Did you listen to the samples this produces ?
